File: gva/server/router/pmocker/time_entry.go
package pmocker

import (
	"github.com/flipped-aurora/gin-vue-admin/server/middleware"
	"github.com/gin-gonic/gin"
)

type TimeEntryRouter struct{}

func (r *TimeEntryRouter) InitTimeEntry(public *gin.RouterGroup, private *gin.RouterGroup) {
	{
		timeEntry := private.Group("timeEntry").Use(middleware.OperationRecord())
		timeEntry.POST("create", apiGroup.TimeEntryApi.CreateTimeEntry)
		timeEntry.PUT("update", apiGroup.TimeEntryApi.UpdateTimeEntry)
		timeEntry.DELETE("delete", apiGroup.TimeEntryApi.DeleteTimeEntry)
		timeEntry.POST("submit", apiGroup.TimeEntryApi.SubmitTimeEntry)
		timeEntry.POST("approve", apiGroup.TimeEntryApi.ApproveTimeEntry)
		timeEntry.POST("reject", apiGroup.TimeEntryApi.RejectTimeEntry)
	}
	{
		timeEntry := private.Group("timeEntry")
		timeEntry.GET("find", apiGroup.TimeEntryApi.FindTimeEntry)
		timeEntry.GET("list", apiGroup.TimeEntryApi.ListTimeEntries)
		timeEntry.GET("utilization", apiGroup.TimeEntryApi.UtilizationTimeEntry)
	}
}

File: gva/web/src/api/pmocker/timeEntry.js
import service from '@/utils/request'

export const createTimeEntry = (data) => service({ url: '/pmocker/timeEntry/create', method: 'post', data })
export const updateTimeEntry = (data) => service({ url: '/pmocker/timeEntry/update', method: 'put', data })
export const deleteTimeEntry = (params) => service({ url: '/pmocker/timeEntry/delete', method: 'delete', params })
export const findTimeEntry = (params) => service({ url: '/pmocker/timeEntry/find', method: 'get', params })
export const listTimeEntries = (params) => service({ url: '/pmocker/timeEntry/list', method: 'get', params })
export const submitTimeEntry = (params) => service({ url: '/pmocker/timeEntry/submit', method: 'post', params })
export const approveTimeEntry = (params) => service({ url: '/pmocker/timeEntry/approve', method: 'post', params })
export const rejectTimeEntry = (params) => service({ url: '/pmocker/timeEntry/reject', method: 'post', params })
export const getUtilization = (params) => service({ url: '/pmocker/timeEntry/utilization', method: 'get', params })
